go-controller/pkg/ovn/policy.go

```diff
@@ -665,16 +666,40 @@ func (oc *Controller) processLocalPodSelectorSetPods(policy *knet.NetworkPolicy,
 			continue
 		}
 
-		// Get the logical port info
 		logicalPort := util.GetLogicalPortName(pod.Namespace, pod.Name)
-		portInfo, err := oc.logicalPortCache.get(logicalPort)
-		// pod is not yet handled
-		// no big deal, we'll get the update when it is.
-		if err != nil {
+		var portInfo *lpInfo
+
+		// Get the logical port info, if the LSP is Scheduled for removal,
+		// wait until it's either gone or re-added to ensure we have the up
+		// to date version in the cache this usually only occurs with stateful-sets
+		//
+		// 24ms is chosen because gomega.Eventually default timeout is 50ms
+		// libovsdb transactions take less than 50ms usually as well so pod create
+		// should be done within a couple iterations
+		retryErr := wait.PollImmediate(24*time.Millisecond, 120*time.Millisecond, func() (bool, error) {
+			var err error
+			// Do not retry if cache get errors out
```

Comment:
why? Couldn't it be the case that the pod handler is slow and hasnt created the port yet? Wouldn't we still want to wait? Oh now I see the previous comment you removed:
		// pod is not yet handled
		// no big deal, we'll get the update when it is.

But I think this comment is not accurate. We do look for pod updates in the network policy pod handler, however the annotation is done first, before the port is created in OVN and added to the portCache. That means we could get an update event for a pod because the annotation was set, end up back in this function, and the port is still not in the cache.

Reply:
You are correct, but in this case specifically I haven't seen that failure mode... Also I'm not really inclined to add another retry here... I think the proper "fix" would be to actually convert this to more of a level driven controller, but for now I'd like to go ahead and fix this bug because it is a pretty proper security hole with stateful-sets